extras/envoy-xds/main.go

```diff
@@ -57,37 +59,15 @@ func main() {
 }
 
 func pollForConfigUpdates(tckr *time.Ticker, snapshotCache cache.SnapshotCache) {
-	lastModTime := int64(0)
 	snapVersion := uint(1)
 	for range tckr.C {
-
-		fi, err := os.Stat(configPath)
-		if err != nil {
-			log.Panicf("Unable to stat given config file: %s. Error: %v", configPath, err)
-		}
-
-		if fi.IsDir() {
-			log.Panicf("Given config path: %s points to a directory. This must be a file.", configPath)
-		}
-
-		currModTime := fi.ModTime().Unix()
-		if currModTime <= lastModTime {
-			continue
-		}
-
-		config, err := ioutil.ReadFile(configPath)
-		if err != nil {
-			log.Panicf("Unable to read given config file: %s. Error: %v", configPath, err)
-		}
-
-		kvs := make(map[string]interface{})
-		err = json.Unmarshal(config, &kvs)
+		envoyConfig, err := discoveryClient.GetEnvoyConfig()
 		if err != nil {
-			log.Panicf("Unable to convert the contents of config file: %s into JSON. Error: %v", configPath, err)
+			log.Printf("Unable to get cluster info: Error: %v", err)
 		}
 
-		if err = pkg.EnvoyDKVConfig(kvs).ComputeAndSetSnapshot(snapVersion, snapshotCache); err != nil {
-			log.Panicf("Unable to compute and set snapshot. Error: %v", err)
+		if err = envoyConfig.ComputeAndSetSnapshot(snapVersion, snapshotCache); err != nil {
+			log.Printf("Unable to compute and set snapshot. Error: %v", err)
```

Comment:
Shouldn't this be panic level only?

Reply:
idea was that now since discovery config is dynamic, we should avoid throwing a panic at runtime (and thus stop the service) and rather wait for the temporary glitch (shard unavailability, config issue, etc) to be resolved